tools/bench/3pc/lua/lua_bench.cpp

```diff
@@ -0,0 +1,350 @@
+// Copyright (c) 2021 MIT Digital Currency Initiative,
+//                    Federal Reserve Bank of Boston
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "3pc/agent/client.hpp"
+#include "3pc/broker/impl.hpp"
+#include "3pc/directory/impl.hpp"
+#include "3pc/runtime_locking_shard/client.hpp"
+#include "3pc/ticket_machine/client.hpp"
+#include "3pc/util.hpp"
+#include "crypto/sha256.h"
+#include "wallet.hpp"
+
+#include <random>
+#include <thread>
+
+auto main(int argc, char** argv) -> int {
+    auto log
+        = std::make_shared<cbdc::logging::log>(cbdc::logging::log_level::warn);
+
+    auto sha2_impl = SHA256AutoDetect();
+    log->info("using sha2: ", sha2_impl);
+
+    if(argc < 2) {
+        log->error("Not enough arguments");
+        return 1;
+    }
+    auto cfg = cbdc::threepc::read_config(argc - 1, argv);
+    if(!cfg.has_value()) {
+        log->error("Error parsing options");
+        return 1;
+    }
+
+    auto args = cbdc::config::get_args(argc, argv);
+    auto n_wallets = std::stoull(args.back());
+    if(n_wallets < 2) {
+        log->error("Must be at least two threads");
+        return 1;
+    }
+
+    log->trace("Connecting to shards");
+    auto shards = std::vector<
+        std::shared_ptr<cbdc::threepc::runtime_locking_shard::interface>>();
+    for(const auto& shard_ep : cfg->m_shard_endpoints) {
+        auto client = std::make_shared<
+            cbdc::threepc::runtime_locking_shard::rpc::client>(
+            std::vector<cbdc::network::endpoint_t>{shard_ep});
+        if(!client->init()) {
+            log->error("Error connecting to shard");
+            return 1;
+        }
+        shards.emplace_back(client);
+    }
+    log->trace("Connected to shards");
+
+    log->trace("Connecting to ticket machine");
+    auto ticketer
+        = std::make_shared<cbdc::threepc::ticket_machine::rpc::client>(
+            std::vector<cbdc::network::endpoint_t>{
+                cfg->m_ticket_machine_endpoints});
+    if(!ticketer->init()) {
+        log->error("Error connecting to ticket machine");
+        return 1;
+    }
+    log->trace("Connected to ticket machine");
+
+    auto directory
+        = std::make_shared<cbdc::threepc::directory::impl>(shards.size());
+    auto broker = std::make_shared<cbdc::threepc::broker::impl>(
+        std::numeric_limits<size_t>::max(),
+        shards,
+        ticketer,
+        directory,
+        log);
+
+    auto pay_contract = cbdc::buffer();
+    pay_contract
+        = cbdc::buffer::from_hex(
+              "1b4c7561540019930d0a1a0a040808785600000000000000000000002877400"
```

Comment:
Currently working on a fix for this

Reply:
As a follow up, is there any reason that comes to mind for store the bytecode in plaintext anywhere? It's fairly easy to generate the bytecode at runtime by leveraging Lua's embedded scripting capabilities and then we don't need to store this massive string anywhere persistently. 

So far I have two solutions: one where this bytecode string is generated during the build process and is stored persistently in a text file, which is then read in at runtime -- and another where the bytecode is generated at runtime (not requiring any external storage, or large static strings). I prefer the second solution but if there's any reason to store the bytecode persistently, I would be interested to hear it.

Reply:
Unless any of the generation would happen during performance measurements, I see no reason to keep around the hex-version if it's just as easy to keep around the source.

Reply:
Unless there is a reason to re-compile every time, compile once and keep the derived (non-git element) persistent file.

Reply:
@AlexRamRam Sorry just now seeing this -- I would say in regards to the bench program it makes sense to interpret the Lua script at runtime because it allows easier interaction with different Lua scripts which could be useful, but in terms of the unit test set up I can see interpreting the script at runtime being unnecessary

Reply:
^^ Also -- not sure if I did a good job explaining this part, but the bytecode is only generated once at runtime (in the setup for either the bench test or unit test) and then it exists statically for the remainder of the test, it is not being regenerated each time the bytecode is used